modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaPlayFrameworkCodegen.java

```diff
@@ -0,0 +1,250 @@
+
+    public static final String TITLE = "title";
+    public static final String CONFIG_PACKAGE = "configPackage";
+    public static final String BASE_PACKAGE = "basePackage";
+    public static final String CONTROLLER_ONLY = "controllerOnly";
+    public static final String SINGLE_CONTENT_TYPES = "singleContentTypes";
+    public static final String RESPONSE_WRAPPER = "responseWrapper";
+    public static final String USE_TAGS = "useTags";
+
+    protected String title = "swagger-petstore";
+    protected String configPackage = "io.swagger.configuration";
+    protected String basePackage = "io.swagger";
+    protected boolean controllerOnly = false;
+    protected boolean singleContentTypes = false;
+    protected String responseWrapper = "";
+    protected boolean useTags = false;
+    protected boolean useBeanValidation = true;
+
+    public JavaPlayFrameworkCodegen() {
+        super();
+        outputFolder = "generated-code/javaPlayFramework";
+        apiTestTemplateFiles.clear();
+        embeddedTemplateDir = templateDir = "JavaPlayFramework";
+        apiPackage = "controllers";
+        modelPackage = "apimodels";
+        invokerPackage = "io.swagger.api";
+        artifactId = "swagger-java-playframework";
+
+        projectFolder = "";
+        sourceFolder = projectFolder + File.separator + "app";
+        projectTestFolder = projectFolder + File.separator + "test";
+        testFolder = projectTestFolder;
+
+        additionalProperties.put(CONFIG_PACKAGE, configPackage);
+        additionalProperties.put(BASE_PACKAGE, basePackage);
+
+        //TODO: Validate this?
+        // play framework uses the jackson lib, why have a parameter for this?
+        additionalProperties.put("jackson", "true");
+
+        cliOptions.add(new CliOption(TITLE, "server title name or client service name"));
+        cliOptions.add(new CliOption(CONFIG_PACKAGE, "configuration package for generated code"));
+        cliOptions.add(new CliOption(BASE_PACKAGE, "base package for generated code"));
+        cliOptions.add(CliOption.newBoolean(CONTROLLER_ONLY, "Whether to generate only API interface stubs without the server files."));
+        cliOptions.add(CliOption.newBoolean(SINGLE_CONTENT_TYPES, "Whether to select only one produces/consumes content-type by operation."));
+        cliOptions.add(new CliOption(RESPONSE_WRAPPER, "wrap the responses in given type (Future,Callable,CompletableFuture,ListenableFuture,DeferredResult,HystrixCommand,RxObservable,RxSingle or fully qualified type)"));
+        cliOptions.add(CliOption.newBoolean(USE_TAGS, "use tags for creating interface and controller classnames"));
+        cliOptions.add(CliOption.newBoolean(USE_BEANVALIDATION, "Use BeanValidation API annotations"));
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    @Override
+    public String getName() {
+        return "javaPlayFramework";
```

Comment:
I would suggest lower case with "-", e.g. java-play-framework
which is the convention used by other generators.